src/Primitives/Search.purs

```diff
@@ -198,6 +219,19 @@ component =
       -- the parent to set the text.
       SearchReceiver i a -> H.modify (updateStore i.render id) *> pure a
 
+      CaptureFocus e a -> a <$ do
+        (Tuple _ st) <- getState
+        let el = either (const Nothing) Just <<< runExcept <<< readHTMLElement <<< toForeign <<< currentTarget <<< focusEventToEvent $ e
```

Comment:
This is pretty long. What about:

```purescript
let el = either (const Nothing) Just 
        <<< runExcept 
        <<< readHTMLElement 
        <<< toForeign 
        <<< currentTarget 
        <<< focusEventToEvent 

H.modify $ seeks _ { inputEl = el e }
H.raise Focused
```

Reply:
Also -- should we check this? Ensure that there isn't already a reference stored? Or would we always want to overwrite just in case?

Reply:
hush!